x-pack/test/api_integration/apis/metrics_ui/services.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { ServicesAPIResponseRT } from '@kbn/infra-plugin/common/http_api/host_details';
+import { decodeOrThrow } from '@kbn/infra-plugin/common/runtime_types';
+import { FtrProviderContext } from './types';
+import { generateServicesData } from './helpers';
+
+const SERVICES_ENDPOINT = '/api/infra/services';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const synthtrace = getService('apmSynthtraceEsClient');
+  describe('GET /infra/services', () => {
+    it('returns no services with no data', async () => {
+      const filters = JSON.stringify({
+        'host.name': 'some-host',
+      });
+      const from = new Date(Date.now() - 1000 * 60 * 2).toISOString();
+      const to = new Date().toISOString();
+      const response = await supertest
+        .get(SERVICES_ENDPOINT)
+        .set({
+          'kbn-xsrf': 'some-xsrf-token',
+        })
+        .query({
+          filters,
+          from,
+          to,
+        })
+        .expect(200);
+
+      const { services } = decodeOrThrow(ServicesAPIResponseRT)(response.body);
```

Comment:
General suggestion: it's VERY useful to have a typed api client. For apm we have [this](https://github.com/sqren/kibana/blob/950c9345464be2cff136154c168b128b010b7318/x-pack/test/apm_api_integration/common/apm_api_supertest.ts#L18) which makes it possible to call REST apis and get typed responses back - no custom parsing needed

<img width="1167" alt="image" src="https://github.com/elastic/kibana/assets/209966/0eb7f693-1974-4611-9db4-47d609010cdc">

Reply:
https://github.com/elastic/kibana/issues/175717